coderd/conversion/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package conversion provides common conversion routines from database types to codersdk types
+package conversion
```

Comment:
I'm wondering if this should be a separate refactor to separate the converting function that has tests... right now all of these functions are _tested_ by coderd code, and I think that's a questionable pattern.

Reply:
I think maybe this should be called `dbconvert` as well and nested in the `database` package.

`conversion` is pretty broad in my mind, and I feel like we want this to be quite narrowly scoped to converting database types to SDK, never the other way around.

Reply:
if we're sure we only want database->SDK, how about `db2sdk` as the package name?  Happy to put it under `database` in the package hierarchy.

As for tests, there are some conversions that have a bit of complexity, like the Status.  I'll whip up a couple test cases for that one tomorrow.

Others are just copying fields, and it feels a bit daft to write a test that just goes through each field and asserts it got copied.  If we're sure we want to exhaustively copy the fields, we could maybe write a non-daft test with reflection, but I'm not sure we generally want to exhaustively copy...

Reply:
I like `db2sdk`, super clear!

I agree with your test-case observations... most would just be silly.